justfile

```diff
@@ -100,14 +100,16 @@ install-precommit:
     test -f $BASE_DIR/.git/hooks/pre-commit || $BIN/pre-commit install
 
 
-# upgrade dev or prod dependencies (specify package to upgrade single package, all by default)
-upgrade env package="": virtualenv
+# compile all requirements.in files (by default only adds/removes packages; `-U` upgrades all; `-P <package>` upgrades one)
+compile-reqs *ARGS: devenv
```

Comment:
(Warning: drive-by comment) It might be worth taking a look at how OpenCodelists does this:

https://github.com/opensafely-core/opencodelists/blob/f7ad55852b63dce37f390a2e15a7056144232aec/justfile#L118-L123

It would be good to be consistent, but if not, then it would be good to know why we're doing something different.

Reply:
I think the Airlock approach is actually better, as alluded to in the associated commit message (a86761b0771a86a63301302055f9ab69b8ef6539), because it reduces the risk of production and development dependencies not being in sync.

That said, the test taken from Airlock also guards against this, so I've left this as in OpenCodelists for now, to reduce the changes in this PR. If we change one, we should change both.